pycheribuild/projects/llvm.py

```diff
@@ -447,6 +447,35 @@ def triple_prefixes_for_binaries(self) -> typing.Iterable[str]:
         return [x + "-" for x in triples]
 
 
+class BuildMorelloLLVM(BuildLLVMMonoRepoBase):
+    repository = GitRepository("https://git.morello-project.org/morello/llvm-project.git")
+    project_name = "morello-llvm-project"
+    target = "morello-llvm"
+    skip_cheri_symlinks = False  # add target-specific symlinks
+    is_sdk_target = True
+    native_install_dir = DefaultInstallDir.MORELLO_SDK
+    supported_architectures = [CompilationTargets.NATIVE]
+
+    @property
+    def triple_prefixes_for_binaries(self) -> typing.Iterable[str]:
+        triples = [
+            CheriBSDMorelloTargetInfo.triple_for_target(CompilationTargets.CHERIBSD_MORELLO_PURECAP, self.config,
+                                                        include_version=False),
+            CheriBSDMorelloTargetInfo.triple_for_target(CompilationTargets.CHERIBSD_MORELLO_PURECAP, self.config,
+                                                        include_version=True),
+            ]
+        return [x + "-" for x in triples]
+
+    def configure(self, **kwargs):
+        # Unless we set the default target triple, CMake will not be able to determine the compiler ID.
+        # The other alternative to fix this problem is to build the host backend. To save build time we do the former.
+        self.add_cmake_options(LLVM_DEFAULT_TARGET_TRIPLE=CheriBSDMorelloTargetInfo.triple_for_target(
+            CompilationTargets.CHERIBSD_MORELLO_PURECAP, self.config, include_version=True))
+        self.add_cmake_options(LLVM_TARGETS_TO_BUILD="AArch64")
+        # self.add_cmake_options(LLVM_TARGETS_TO_BUILD="AArch64;host")
```

Comment:
I've seen issues with some third party software where the native backend was missing, e.g. `cc` fails so it doesn't use it.

Reply:
Yes, I think lots of autotools, etc. configure checks are totally broken and check the wrong thing. Probably best not to have to debug their broken stuff and just build the native backend instead.